File: Easy leetcode/2894. Divisible and Non-divisible Sums Difference.py
class Solution:
    def differenceOfSums(self, n, m):
        totalSum = n * (n + 1) // 2
        k = n // m
        sumMultiplesOfM = m * k * (k + 1) // 2
        sumNotMultiplesOfM = totalSum - sumMultiplesOfM
        return sumNotMultiplesOfM - sumMultiplesOfM

File: Easy leetcode/2894. Divisible and Non-divisible Sums Difference.java
class Solution {
    public int differenceOfSums(int n, int m) {
        int totalSum = n * (n + 1) / 2;
        int k = n / m;
        int sumMultiplesOfM = m * k * (k + 1) / 2;
        int sumNotMultiplesOfM = totalSum - sumMultiplesOfM;
        return sumNotMultiplesOfM - sumMultiplesOfM;
    }

    public static void main(String[] args) {
        java.util.Scanner sc = new java.util.Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        Solution sol = new Solution();
        System.out.println(sol.differenceOfSums(n, m));
    }
}

